docs/metrics/selfcheckgpt.md

```diff
@@ -0,0 +1,157 @@
+Language Language Model (LLM) has inherent knowledge of facts and if prompted multiple times should be output a
+similar/consistent response. Existing fact-checking methods require output token-level probabilities which may not be
+accessible. As such, [SelfCheckGPT](https://arxiv.org/abs/2303.08896) presents a black-box zero-resource solution that
+only requires text-based responses to evaluate for hallucinations.
+
+![SelfCheckGPT Image](../assets/images/selfcheck_qa_prompt.png)
+*[Image from SelfCheckGPT repo](https://github.com/potsawee/selfcheckgpt)*
+
+To compute the SelfCheck Consistency Score, we focus on a few SelfCheckGPT variants:
+
+1. SelfCheckGPT with BERTScore
+3. SelfCheckGPT with $n$-gram
+5. SelfCheckGPT with LLM Prompt
+
+## Implementation Details
+
+SelfCheckGPT contains detailed and comprehensive usage details in their [code repository](https://github.com/potsawee/selfcheckgpt).
+However, we present some simple example use cases for each variant below using an example from the TruthfulQA dataset
+
+```
+Answer: The seeds and the white membrane, also known as the pith, are the spiciest parts of a chili pepper.
+Main Answer: The spiciest part of a chili pepper is the white pith, also known as the placenta, that directly surrounds the seeds.
+```
+
+### BERTScore-based Consistency Score
+
+Given the $i$-th sentence of a response $R$, SelfCheckGPT with BERTScore finds the average BERTScore with the most
+similar sentence from each of the $N$ sample responses. The idea is that if the same information appears in multiple
+sample responses then it should be factually correct.
+
+!!! info "Guide: BERTScore"
+
+    Read the [BERTScore](./bertscore.md) guide if you're not familiar.
+
+
+#### Code Example
+
+The score we compute for each sentence will be between `[0.0, 1.0]` with a higher value suggesting that the response may
+be non-factual.
+
+```py
+selfcheck_bertscore = SelfCheckBERTScore(rescale_with_baseline=True)
+
+sent_scores_bertscore = selfcheck_bertscore.predict(
+    sentences = [main_answer], # list of sentences
+    sampled_passages = [answer], # list of sampled passages
+)
+print(sent_scores_bertscore)
+```
+
+```
+[0.31471425]
+```
+
+### $n$-Gram-based Consistency Score
+
+SelfCheckGPT with $n$-gram trains a simple $n$-gram model using the $N$ sample responses as well as the response $R$ to
+compute token-level probabilities for tokens in each sentence from $R$. The score is computed by the following formula:
+
+$$
+S_{n\text{-gram}}^\text{Avg}(i) = -\frac{1}{J} \sum_{j} \log{\tilde{p}_{ij}}
+$$
+
+where $\tilde{p}_{ij}$ is the probability output from the $n$-gram model of the $j$-th token from $i$-th sentence in $R$.
+
+#### Code Example
+
+The scores we compute are at sentence- and document-level where values lie between `[0.0, +inf)` with a higher value
+suggesting that the response may be non-factual.
+
+```py
+selfcheck_ngram = SelfCheckNgram(n=1) # n=1 means Unigram, n=2 means Bigram, etc.
```

Comment:
Same thing here.  You should be able to copy the block and run it without any issues.